### ISSUE TITLE
Prisma deploy schema generation post-deploy hook failing cause of 2 versions of GraphQL using both graphql-yoga & graphqlgen

### ISSUE THREAD

**Issue by kevinmarrec** [prisma/graphqlgen#186]
## Description
`prisma generate` called as a post-deploy hook by `prisma deploy` fails (see screenshot below)

## Steps to reproduce
Having this `package.json` :
```js
{
  "name": "prisma-typescript",
  "version": "0.1.0",
  "author": "Kevin Marrec",
  "private": true,
  "scripts": {
    "lint": "tslint --project .",
    "lint:fix": "tslint --project . --fix",
    "start": "nodemon src/index.ts",
    "debug": "nodemon --inspect src/index.ts"
  },
  "nodemonConfig": {
    "exec": "ts-node -r dotenv/config",
    "watch": "src",
    "ext": "ts,graphql"
  },
  "dependencies": {
    "bcrypt": "^3.0.2",
    "graphql-yoga": "^1.16.7",
    "jsonwebtoken": "^8.3.0",
    "prisma-client-lib": "^1.18.1"
  },
  "devDependencies": {
    "dotenv": "^6.0.0",
    "graphqlgen": "^0.2.2",
    "nodemon": "^1.18.4",
    "prisma": "^1.18.1",
    "ts-node": "^7.0.1",
    "tslint": "^5.11.0",
    "tslint-config-standard": "^8.0.1",
    "typescript": "^3.1.3"
  }
}
```

## Expected results
No errors / Schema generated correctly

## Actual results
It seems there is a bug having 2 versions of `graphql` from both `graphql-yoga` & `graphqlgen` during prisma generation.

![screenshot_1](https://user-images.githubusercontent.com/25272043/47283789-07ce7c80-d5e5-11e8-9aa4-72cfab63a719.png)

yarn.lock (only the weird part with 2 different `graphql` version definitions) :
```
graphql-yoga@^1.16.7:
  version "1.16.7"
  resolved "https://registry.yarnpkg.com/graphql-yoga/-/graphql-yoga-1.16.7.tgz#05bdfadf9a4623b0409bee3c5b75b7b3f7249e6f"
  integrity sha512-N82SmBpKyDD7GXXo2NbVBMPffbxFWD+YOf/Vqx3Kujj/hNOX+0s/7G+EPp4TgJZqHQjcOw5TqQ0kOANO8w6Kmg==
  dependencies:
    "@types/cors" "^2.8.4"
    "@types/express" "^4.11.1"
    "@types/graphql" "^14.0.0"
    "@types/graphql-deduplicator" "^2.0.0"
    "@types/zen-observable" "^0.5.3"
    apollo-server-express "^1.3.6"
    apollo-server-lambda "1.3.6"
    apollo-upload-server "^5.0.0"
    aws-lambda "^0.1.2"
    body-parser-graphql "1.1.0"
    cors "^2.8.4"
    express "^4.16.3"
    graphql "^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0"
    graphql-deduplicator "^2.0.1"
    graphql-import "^0.7.0"
    graphql-middleware "1.7.7"
    graphql-playground-middleware-express "1.7.6"
    graphql-playground-middleware-lambda "1.7.6"
    graphql-subscriptions "^0.5.8"
    graphql-tools "^4.0.0"
    subscriptions-transport-ws "^0.9.8"

graphql@0.13:
  version "0.13.2"
  resolved "https://registry.yarnpkg.com/graphql/-/graphql-0.13.2.tgz#4c740ae3c222823e7004096f832e7b93b2108270"
  integrity sha512-QZ5BL8ZO/B20VA8APauGBg3GyEgZ19eduvpLWoq5x7gMmWnHoy8rlQWPLmWgFvo1yNgjSEFMesmS4R6pPr7xog==
  dependencies:
    iterall "^1.2.1"

"graphql@^0.11.0 || ^0.12.0 || ^0.13.0 || ^14.0.0", "graphql@^0.13.0 || ^14.0.0", graphql@^14.0.2:
  version "14.0.2"
  resolved "https://registry.yarnpkg.com/graphql/-/graphql-14.0.2.tgz#7dded337a4c3fd2d075692323384034b357f5650"
  integrity sha512-gUC4YYsaiSJT1h40krG3J+USGlwhzNTXSb4IOZljn9ag5Tj+RkoXrWp+Kh7WyE3t1NCfab5kzCuxBIvOMERMXw==
  dependencies:
    iterall "^1.2.2"

graphqlgen-json-schema@0.2.5:
  version "0.2.5"
  resolved "https://registry.yarnpkg.com/graphqlgen-json-schema/-/graphqlgen-json-schema-0.2.5.tgz#a458c55ecb99926b3ee8cf4eacb97915b7085c9a"
  integrity sha512-p5Oq0aigusiSYdpuGwST7mfHk91hrzL27HaOqBW2Y7Y0GgiMH7TJYa/WIqf95uH8bSyRzvBzawTQF7/WqnymPQ==

graphqlgen@^0.2.2:
  version "0.2.6"
  resolved "https://registry.yarnpkg.com/graphqlgen/-/graphqlgen-0.2.6.tgz#6adef254ca8ad9692ae4b5ce418fc8ab4889b4fd"
  integrity sha512-Vyd/sR6DvGlXjXgMqziSmmn0421pOENYUkIbryUo+JR4ZpsyrLwz7wITO+GmrwyJx7dVb7ZGYg1aeDV+rS0TvQ==
  dependencies:
    ajv "^6.5.4"
    camelcase "5.0.0"
    chalk "2.4.1"
    graphql "^0.13.0 || ^14.0.0"
    graphql-import "0.7.1"
    graphqlgen-json-schema "0.2.5"
    js-yaml "3.12.0"
    mkdirp "0.5.1"
    prettier "1.14.3"
    reason "3.3.4"
    rimraf "2.6.2"
    typescript "3.1.3"
    yargs "12.0.2"
```

## Versions
- graphql-yoga: 1.16.7
- graphqlgen: 0.2.6

**Comment by schickling** [comment]
Can you confirm this also happens when running `prisma generate` manually (without running `prisma deploy`)?

**Comment by kevinmarrec** [comment]
@schickling It was happening, yes.

The problem is that `prisma-lib-core` (dependency of `prisma`) use `graphql@0.13`, and I had `prisma` as devDependecy.

I fixed it removing `prisma` dependency and install it globally but I would want to find a way to have it as a devDependency and use `yarn prisma`.

Somtimes you just don't have access to sudo privileges so you can't `sudo yarn global add prisma` to have the `prisma` binary.

**Comment by timsuchanek** [comment]
Thanks @kevinmarrec for the information! That means for us that we have to upgrade the complete prisma cli to graphql@14, which is a bigger tasks, as there are many dependencies.
We want to tackle this next week!

One alternative workaround right now could be `npx` though :) So by using `npx prisma` in your package.json script, you could also circumvent the global sudo stuff.

**Comment by kevinmarrec** [comment]
@timsuchanek It will be fixed in Prisma 1.21 ?
https://github.com/prisma/prisma/releases/tag/1.21.0-beta
https://github.com/prisma/prisma/commit/8eb1623ad7ac2f676bcc029b933884b778fa3d28

**Comment by kevinmarrec** [comment]
Updating to last Prisma version (at the time I write `1.20.1`) fixes this issue.
